### DISCUSSION TITLE
State of ReConServer

### DISCUSSION THREAD

**Discussion by TheQue42** [resiprocate/resiprocate#207] [General]
Hi, 

I just started looking into using resiprocate as a base for a conference server of sorts, so I am fully busy with trying to wrap my head around the architecture, and the various stack layers etc.

The recon, and reConServer stuff looks extremely interesting, but I cant figure out the difference between the code in apps/reConServer/ directory, or the separate https://github.com/resiprocate/reConServer repo?

How up2date is the reconserver code, in either of them?

**Comment by sgodin** [comment]
I can't speak to the reConServer code, I'll let Daniel Pocock field that
question, but I believe it is a B2BUA based on recon.

I will point out some sources of other information for recon though:

   - Some docs here.  A little dated, but mostly accurate:
   https://github.com/resiprocate/resiprocate/tree/master/resip/recon/doc
   - There is a Music On Hold and Park server based off of recon/sipXtapi
   in the code repo:
      -
      https://github.com/resiprocate/resiprocate/tree/master/resip/recon/MOHParkServer
      -
      https://github.com/resiprocate/resiprocate/blob/master/resip/recon/MOHParkServer/doc/MOHParkServer_User_Documentation.pdf
      - Unfortunately the main sipXtapi SVN repo (
   http://sipxsvn.sipez.com/rep/sipX/main) is currently down.  I'm working
   with the project owner to try to get this repo up and running with the
   latest code into GitHub.  In the meantime, this mirror (
   https://github.com/sipXtapi/sipXtapi-svn-mirror ) has the most up to
   date code currently available publically.  Unfortunately it is a few months
   out of date and the latest recon code likely won't fully compile with it.
   We are working to get all the latest sipXtapi changes into a public repo,
   ASAP.
   - I have recently been adding Instant Message support to recon.  This
   work is not reflected in the wiki or the documentation yet.  Reviewing
   recent commit comments from recon is your best visibility into this at the
   moment, if you are interested.
   - There is also some new work currently underway to integrate webrtc via
   Kurento's webrtc implementation into recon as another option to replace
   sipXtapi media.  Daniel has posted to the mailing lists on this.

Regards,

--
Scott Godin
SIP Spectrum, Inc.
www.sipspectrum.com

**Reply by TheQue42** [reply]
Thanks for a quick reply! The MOH Server was a good pointer that I've missed how valuable it will be. Yes, I have noticed the issues with the media-libraries and that Kurento might be on its way in.
It not such an issue for me though, since the customer I am working for have their own media-library they want me to use, so I'll have to try to separate the interfaces between signalling and media myself.

I am however somewhat rusty on classic Linux C++ development, since my major experience was done building the SIP stack for the Ericsson IMS system, with a ton of proprietary tools, utilities and dev-env.

So I'll apologise in advance if I accidentally ask questions that might be a bit n00b-isch :-)
